src/App.tsx

```diff
@@ -15,48 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.LENGTH:
+      visibleGoods.sort((curr, next) => curr.length - next.length);
+      break;
+    case SortType.ALPABET:
+      visibleGoods.sort((curr, next) => curr.localeCompare(next));
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortTypes, setSortTypes] = useState<SortType>(SortType.NONE);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+
+  const resetState = () => {
+    setSortTypes(SortType.NONE);
+    setIsReverse(false);
+  };
+
+  const getOrderedGoods = getReorderedGoods(
+    goodsFromServer,
+    sortTypes,
+    isReverse,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-info',
+            { 'is-light': sortTypes !== SortType.ALPABET })}
+          onClick={() => setSortTypes(SortType.ALPABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button', 'is-success',
+            { 'is-light': sortTypes !== SortType.LENGTH })}
+          onClick={() => setSortTypes(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button', 'is-warning',
+            { 'is-light': !isReverse })}
+          onClick={() => setIsReverse(!isReverse)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReverse || sortTypes !== SortType.NONE) && (
```

Comment:
you didn't fix previous comment
let's create separate variable with understandable name, where you can store the result for this check
and let's use this variable here